File: binaryArray.py
def countOnes(arr,low,high):
      
    if high>=low:

        mid = low + (high-low)//2

        if ((mid == high or arr[mid+1]==0) and (arr[mid]==1)):
            return mid+2

        if arr[mid]==1:
            return countOnes(arr, (mid+1), high)
              
        return countOnes(arr, low, mid-1)
      
    return 0

arr=[1, 1, 0, 1, 1, 1]
print ("Output :",countOnes(arr, 0 , len(arr)-3))

File: binaryArray.php
<?php

function binarySearch($arr, $l, $r, $x)
{
if ($r >= $l)
{
        $mid = ceil($l + ($r - $l) / 2);

        if ($arr[$mid] == $x || $arr[$mid+1]==0) && ($arr[$mid]==1)
            return floor($mid+2);

        if ($arr[$mid] == 1) 
            return binarySearch($arr, $l, 
                                $mid + 1, $x);

        return binarySearch($arr, $mid - 1, 
                            $r, $x);
}

return 0;
}

$arr = array(2, 3, 4, 10, 40);

if(($result == -1))
echo "Element is not present in array";
else
echo "Element is present at index ",
                            $result;

?>

?>